crates/rpc/rpc-types-compat/src/transaction.rs

```diff
@@ -46,3 +50,86 @@ pub trait TransactionCompat: Send + Sync + Unpin + Clone + Debug {
         request: TransactionRequest,
     ) -> Result<TxTy<Self::Primitives>, Self::Error>;
 }
+
+/// Converts `self` into `T`.
+pub trait IntoRpcTx<T> {
+    /// Performs the conversion.
+    fn into_rpc_tx(self, tx_info: TransactionInfo) -> T;
+}
+
+/// Converts [`TransactionRequest`] into `Self`.
+pub trait FromTxReq {
+    /// Performs the conversion.
+    fn from_tx_req(request: TransactionRequest) -> Result<Self, ValueError<TransactionRequest>>
```

Comment:
since this is only for simulateV1 which has a dedicated signature, we could add this context here either in docs or in the function name?

or we add a function that also accepts the signature that should be used here